lib/server.js

```diff
@@ -1,115 +1,155 @@
-    async.series([
-      nginxConf.bind(null, app),
-      app._nginxCmd.bind(app, 'reload')
-    ], function(err) {
-      if (err) return callback(err);
-
-      app.emit('reloaded');
-      console.log(
-        'StrongLoop Nginx Controller listening at: http://%s:%s',
-        app.get('host'),
-        app.get('port')
-      );
-      callback();
-    });
+  if (options.apiEndpoint) {
+    var apiEndpoint = url.parse(options.apiEndpoint);
+    app.set('host', apiEndpoint.hostname);
+    app.set('port', apiEndpoint.port);
+    app._disableApi = false;
   }
-  app._reloadNginx = _reloadNginx;
 
-  function _configureNginx(callback) {
-    nginxConf(app, callback);
-  }
-  app._configureNginx = _configureNginx;
-
-  /**
-   * start/stop/reload Nginx daemon.
-   * @private
-   */
-  function _nginxCmd(action, callback) {
-    var nginxConfdir = path.join(app.get('baseDir'), 'nginx');
-    var configFile = path.join(nginxConfdir, 'nginx.conf');
-
-    var cmd = [app.get('nginxPath'), '-p', nginxConfdir, '-c', configFile];
-    if (action !== 'start') cmd.push('-s', action);
-
-    debug('CMD: %s', cmd.join(' '));
-    childProcess.exec(cmd.join(' '), function(err, stdout, stderr) {
-      if (err) debug('error', err);
-      debug('stdout: <\n%s>', stdout);
-      debug('stderr: <\n%s>', stderr);
-      callback(err);
-    });
-  }
-  app._nginxCmd = _nginxCmd;
-
-  function start(callback) {
-    var controlAddr;
-    async.series([
-      function(done) {
-        app.listen(function() {
-          controlAddr = this.address();
-          debug('control listened on:', this.address());
-          return done();
-        });
-      },
-      app._configureNginx.bind(app),
-      app._nginxCmd.bind(app, 'start')
-    ], function(err) {
-      if (err) {
-        console.log('Error starting: ', err.message);
-        throw err;
-      }
+  app.use(loopback.favicon());
+  app.use(loopback.compress());
 
-      if (debug.enabled)
-        app.emit('started'); // Enable Explorer
+  app.dataSource('db', {
+    name: 'db',
+    connector: 'memory',
+    file: path.join(options.baseDir, 'routes.json'),
+  });
 
-      app.emit('listening', controlAddr);
+  boot(app, __dirname);
+  app.use(loopback.urlNotFound());
+  app.use(loopback.errorHandler());
 
-      if (callback) return callback();
+  app.start = function start(callback) {
+    var self = this;
+    var tasks = [];
+    if (!self._disableApi) {
+      tasks.push(function(callback) {
+        app.listen(function() {
+          self._apiAddress = this.address();
+          debug('API listening on: ', self._apiAddress);
+          callback();
+        });
+      });
+    }
+    tasks.push(self._nginx.start.bind(self._nginx));
+    if (self._controlUri) {
+      tasks.push(self._initControlChannel.bind(self));
+    }
+    async.series(tasks, function(err) {
+      if (err && callback) return callback(err);
+      if (self._apiAddress) self.emit('listening', self._apiAddress);
```

Comment:
Actually, it would fix a bug.. if no callback if given, this function doesn't actually treat the error as an error - it falls through to the success case.